src/arch/x86_64/bits.zig

```diff
@@ -145,7 +143,7 @@ pub const Register = enum(u7) {
 // zig fmt: on
 
 /// These registers belong to the called function.
-pub const callee_preserved_regs = [_]Register{ .rax, .rcx, .rdx, .rsi, .rdi, .r8, .r9, .r10, .r11 };
+pub const callee_preserved_regs = [_]Register{ .rcx, .rsi, .rdi, .r8, .r9, .r10, .r11 };
```

Comment:
According to [this guide](http://cons.mit.edu/sp17/x86-64-architecture-guide.html), callee saved registers are `.rbx, .rbp, .r12, .r13, .r14, .r15`, while the rest belongs to the callee including `.rax` and `.rdx`. Is there a reason why you removed them from the list?

Reply:
I was following the guidance of @joachimschmidt557 https://discord.com/channels/605571803288698900/785499283368706060/908118081434435614 , and it fixed the test case in #10138 . It also didn't make sense how if they needed to be pushed and popped in every function that they could be returned. What do you think?

Reply:
Ah, I see. My suggestion would be to go with your changes for the moment, but drop a TODO comment to figure it out in such a way so that we adhere to the amd64 manual (i.e., having the full registers set).

Reply:
Ok, thanks for the suggestion!

Reply:
I don't see how the information provided in the guide contradicts removing `rax` and `rdx`. The guide describes these registers as not "saved across calls", which is what I understand to be a `callee_preserved_register`. The definition I currently have for a `callee_preserved_register` is that such a register is not affected in any way when calling another function (whether the other function itself overwrites it or when this register is used as a parameter to that function and is thus overwritten etc.). As far as I can tell, this is also the definition the guide uses.

Reply:
Mhm, that sounds about right. My argument here was to investigate why there is a need to explicitly remove two registers from the list thus making them not used automatically by the register allocator especially if they are callee preserved.